[BUG] Compilation of FW with 2 extruders.

### Did you test the latest `bugfix-2.1.x` code?

Yes, and the problem still exists.

### Bug Description

I have a printer with 2 nozzles. I regularly update to new versions.
Now the compilation ends with an error.
I found that it brought the error
https://github.com/MarlinFirmware/Marlin/pull/26152


### Bug Timeline

24.1.2024

### Expected behavior

Compilation of FW

### Actual behavior

Compilation error

![image](https://github.com/MarlinFirmware/Marlin/assets/46979052/2c928ef6-a102-4f34-97da-065a5a79c272)



After commenting this line 
u8g.drawBitmapP(_MAX(0, STATUS_HOTEND_X(heater_id) - 6), STATUS_HEATERS_Y + 3, 1, 5, status_active_extruder_indicator_bmp);
in the status_screen_DOGM.cpp file, the compilation will succeed



### Steps to Reproduce

_No response_

### Version of Marlin Firmware

bugfix 2.1.x

### Printer model

Ender3 - modified

### Electronics

SKR V1.4 Turbo

### LCD/Controller

BTT TFT 3.5 E3 V3.0

### Other add-ons

Print head - pendulum

### Bed Leveling

ABL Linear grid

### Your Slicer

Cura

### Host Software

None

### Don't forget to include

- [X] A ZIP file containing your `Configuration.h` and `Configuration_adv.h`.

### Additional information & file uploads

[config.zip](https://github.com/MarlinFirmware/Marlin/files/14470594/config.zip)

Your config does not compile with stock Marlin, so please attach your modified pins file / any other modified files:

```prolog
Marlin/src/HAL/shared/../../inc/SanityCheck.h:714:6: error: #error "SERVO1_PIN must be defined for your SWITCHING_NOZZLE."
  714 |     #error "SERVO1_PIN must be defined for your SWITCHING_NOZZLE."
      |      ^~~~~
Marlin/src/HAL/shared/../../inc/SanityCheck.h:756:6: error: #error "SERVO1_PIN must be defined for your SWITCHING_EXTRUDER."
  756 |     #error "SERVO1_PIN must be defined for your SWITCHING_EXTRUDER."
      |      ^~~~~
Marlin/src/HAL/LPC1768/../../inc/SanityCheck.h:964:26: error: static assertion failed: NUM_SERVOS (or some servo index) is too large. The selected board only has 1 servos.
```

--- a/Marlin/src/lcd/dogm/status_screen_DOGM.cpp
+++ b/Marlin/src/lcd/dogm/status_screen_DOGM.cpp
@@ -323,7 +323,7 @@ FORCE_INLINE void _draw_centered_temp(const celsius_t temp, const uint8_t tx, co
 
       } // PAGE_CONTAINS
 
-      #if HAS_MULTI_EXTRUDER && NONE(SLIM_LCD_MENUS, STATUS_HOTEND_NUMBERLESS)
+      #if HAS_MULTI_EXTRUDER && NONE(SLIM_LCD_MENUS, STATUS_HOTEND_NUMBERLESS, SINGLENOZZLE)
         if (active_extruder == heater_id)
           u8g.drawBitmapP(_MAX(0, STATUS_HOTEND_X(heater_id) - 6), STATUS_HEATERS_Y + 3, 1, 5, status_active_extruder_indicator_bmp);
       #endif

--- a/Marlin/src/lcd/dogm/status_screen_DOGM.cpp
+++ b/Marlin/src/lcd/dogm/status_screen_DOGM.cpp
@@ -323,7 +323,7 @@ FORCE_INLINE void _draw_centered_temp(const celsius_t temp, const uint8_t tx, co
 
       } // PAGE_CONTAINS
 
-      #if HAS_MULTI_EXTRUDER && DISABLED(SLIM_LCD_MENUS)
+      #if HAS_MULTI_EXTRUDER && NONE(SLIM_LCD_MENUS, STATUS_HOTEND_NUMBERLESS)
         if (active_extruder == heater_id)
           u8g.drawBitmapP(_MAX(0, STATUS_HOTEND_X(heater_id) - 6), STATUS_HEATERS_Y + 3, 1, 5, status_active_extruder_indicator_bmp);
       #endif

